hicommonwealth/commonwealth
# Remove chain node from topics + Fix ERC20 voting

## Link to Issue
Closes: #9511 

## Description of Changes
- Removed chain node ID from topic entity
- Updated vote weight calculate func to use community chain node for both stake and ERC20
- Changed vote weight columns from integer type to numeric to support large digit numbers
- Changed UI to use string for vote weight columns + BigNumber for math operations
- Changed vote multiplier column to support decimals

## Test Plan
- Set env `FLAG_FARCASTER_CONTEST=false` (needs to be disabled)
- Set env `FLAG_WEIGHTED_TOPICS=true`
- Run migration
- ERC20
  - Create a topic with ERC20 weighted voting
    - NOTE: The UI may be broken when creating an ERC20 topic– if so, use API or modify DB topic directly
    - If you have CMN test token, use `0x429ae85883f82203D736e8fc203A455990745ca1` for the address
  - Vote on thread within topic, confirm upvote weight equals token balance
- Stake
  - Create a topic with staked voting
  - Vote on thread within topic, confirm upvote weight equals stake balance
- Unweighted
  - Create a topic without weighted voting
  - Vote on thread within topic, confirm weight == 1

## Deployment Plan
Enable feature flag for weighted voting: `FLAG_WEIGHTED_TOPICS`

## Other Considerations
N/A
